flink-connector-sqlserver-cdc/src/main/java/com/ververica/cdc/connectors/sqlserver/SqlServerSource.java

```diff
@@ -0,0 +1,155 @@
+            this.username = username;
+            return this;
+        }
+
+        /** Password to use when connecting to the SQL Server database server. */
+        public Builder<T> password(String password) {
+            this.password = password;
+            return this;
+        }
+
+        /** The Debezium Sqlserver connector properties. For example, "snapshot.mode". */
+        public Builder<T> debeziumProperties(Properties properties) {
+            this.dbzProperties = properties;
+            return this;
+        }
+
+        /**
+         * The deserializer used to convert from consumed {@link
+         * org.apache.kafka.connect.source.SourceRecord}.
+         */
+        public Builder<T> deserializer(DebeziumDeserializationSchema<T> deserializer) {
+            this.deserializer = deserializer;
+            return this;
+        }
+
+        public DebeziumSourceFunction<T> build() {
+            Properties props = new Properties();
+            props.setProperty("connector.class", SqlServerConnector.class.getCanonicalName());
+            // hard code server name, because we don't need to distinguish it, docs:
+            // Logical name that identifies and provides a namespace for the SQL Server database
+            // server that you want Debezium to capture. The logical name should be unique across
+            // all other connectors, since it is used as a prefix for all Kafka topic names
+            // emanating from this connector. Only alphanumeric characters and underscores should be
+            // used.
+            props.setProperty("database.server.name", DATABASE_SERVER_NAME);
+            props.setProperty("database.hostname", checkNotNull(hostname));
+            props.setProperty("database.user", checkNotNull(username));
+            props.setProperty("database.password", checkNotNull(password));
+            props.setProperty("database.port", String.valueOf(port));
+            props.setProperty("database.history.skip.unparseable.ddl", String.valueOf(true));
+            props.setProperty("database.dbname", checkNotNull(database));
+
+            if (tableList != null) {
+                props.setProperty("table.include.list", String.join(",", tableList));
+            }
+
+            switch (startupOptions.startupMode) {
+                case INITIAL:
+                    props.setProperty("snapshot.mode", "initial");
+                    break;
+                case INITIAL_ONLY:
+                    props.setProperty("snapshot.mode", "initial_only");
+                    break;
+                case SCHEMA_ONLY:
+                    props.setProperty("snapshot.mode", "schema_only");
+                    break;
+                default:
+                    throw new UnsupportedOperationException();
+            }
+
+            if (dbzProperties != null) {
+                dbzProperties.forEach(props::put);
+            }
+
+            return new DebeziumSourceFunction<>(
```

Comment:
Add a validator to verify whether the database version matches and whether CDC is enabled?

Reply:
> Add a validator to verify whether the database version matches and whether CDC is enabled?

Ok,It should  add cdc enabled validation. Thanks for your advice.